packages/node/src/integrations/localvariables.ts

```diff
@@ -4,14 +4,39 @@ import { LRUMap } from 'lru_map';
 
 import type { NodeClientOptions } from '../types';
 
+type Variables = Record<string, unknown>;
+type OnPauseEvent = InspectorNotification<Debugger.PausedEventDataType>;
 export interface DebugSession {
   /** Configures and connects to the debug session */
-  configureAndConnect(
-    onPause: (message: InspectorNotification<Debugger.PausedEventDataType>) => void,
-    captureAll: boolean,
-  ): void;
+  configureAndConnect(onPause: (message: OnPauseEvent, complete: () => void) => void, captureAll: boolean): void;
   /** Gets local variables for an objectId */
-  getLocalVariables(objectId: string): Promise<Record<string, unknown>>;
+  getLocalVariables(objectId: string, callback: (vars: Variables) => void): void;
+}
+
+type Next<T> = (result: T) => void;
+type Add<T> = (fn: Next<T>) => void;
+type CallbackWrapper<T> = { add: Add<T>; next: Next<T> };
+
+function createCallbackList<T>(complete: Next<T>): CallbackWrapper<T> {
+  // A collection of callbacks to be executed last to first
+  const callbacks: Next<T>[] = [complete];
+
+  function add(fn: Next<T>): void {
+    callbacks.push(fn);
+  }
+
+  function next(result: T): void {
+    const popped = callbacks.pop() || complete;
+
+    try {
+      popped(result);
+    } catch (_) {
+      // If there is an error, we still want to call the complete callback
+      complete(result);
```

Comment:
Should we clear the callbacks list here? Otherwise there is a risk of `complete` being called multiple times right?